mgnify_pipelines_toolkit/analysis/assembly/study_summary_generator.py

```diff
@@ -285,7 +288,17 @@ def generate_functional_summary(
     M00163	83.33	Photosystem I	Pathway modules; Energy metabolism; Photosynthesis	K02689,K02690,K02691,K02692,K02694	K02693
     M00615	50.0	Nitrate assimilation	Signature modules; Module set; Metabolic capacity	K02575	M00531
     """
-    check_files_exist(list(file_dict.values()))
+    try:
+        check_files_exist(list(file_dict.values()))
+    except FileNotFoundError as e:
+        if allow_missing:
+            logging.warning(
+                f"One of the expected files is missing, but this is allowed for {label}."
+            )
+            logging.warning(e)
+            return
+        else:
+            raise e
```

Comment:
I would make the end more concise:
```suggestion
        if allow_missing:
            logging.warning(
                f"One of the expected files is missing, but this is allowed for {label}."
            )
            logging.warning(e)
            return
        raise
```